cub/cub/util_type.cuh

```diff
@@ -262,10 +262,12 @@ struct FutureValue
 {
   using value_type    = T;
   using iterator_type = IterT;
+
   explicit _CCCL_HOST_DEVICE _CCCL_FORCEINLINE FutureValue(IterT iter)
       : m_iter(iter)
   {}
-  _CCCL_HOST_DEVICE _CCCL_FORCEINLINE operator T()
+
+  _CCCL_HOST_DEVICE _CCCL_FORCEINLINE operator const T&() const noexcept
   {
     return *m_iter;
   }
```

Comment:
I retested the `offset_iterator` from #4073 and I think I can even step back to returning a value:
```suggestion
  _CCCL_HOST_DEVICE _CCCL_FORCEINLINE operator T() const noexcept
  {
    return *m_iter;
  }
```
Should we do that?

Reply:
Mh, I'd still have the same concern that users might accidentally dereference on the host was is a device-accessible iterator. But if you feel strongly that we need it, it's ok to go ahead.